src/main/java/useCases/userLogin/UserLoginInteractor.java

```diff
@@ -0,0 +1,32 @@
+package useCases.userLogin;
+
+import entities.LoggedInUserSingleton;
+import intefaceAdapters.userLogin.FindUser;
+
+import java.io.IOException;
+
+public class UserLoginInteractor implements UserLoginInputBoundary {
+    private final UserLoginDsGateway userLoginDsGateway;
+    private final UserLoginPresenter userLoginPresenter;
+
+    public UserLoginInteractor(UserLoginPresenter userLoginPresenter) {
+        try {
+            userLoginDsGateway = new FindUser("./users.csv");
+        } catch (IOException e) {
+            throw new RuntimeException("Could not find file");
+        }
+        this.userLoginPresenter = userLoginPresenter;
+    }
+
+
+    @Override
+    public void logInUser(UserLoginRequestModel requestModel) {
+        if (userLoginDsGateway.validLogin(requestModel.getEmail(), requestModel.getPassword())) {
+            UserLoginResponseModel userLoginResponseModel = new UserLoginResponseModel("Welcome back to Ryde!", requestModel.getEmail());
+            LoggedInUserSingleton.init(requestModel.getEmail());
+            userLoginPresenter.loginSuccess(userLoginResponseModel);
+        } else {
```

Comment:
This piece is kind of hard to read. I would recommend extracting the email and password as separate variables from the request model and then inputting them as parameters.